### PR TITLE
More explicit API for converting search sorted results into indices

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

File: vortex-array/src/compute/search_sorted.rs [hunk 1]
```diff
@@ -23,26 +23,59 @@ impl Display for SearchSortedSide {
...
             Self::NotFound(i) => i,
         }
     }
+
+    /// Convert search result into an index suitable for index 0 offset ends arrays, like ChunkArray ends
+    pub fn to_offset_ends_index(self, len: usize) -> usize {
+        match self {
+            SearchResult::Found(i) => {
+                if i == len {
+                    i - 1
```
Comment (lwwmanning) [inline vortex-array/src/compute/search_sorted.rs:59]: maybe more succinctly expressed as `min(i, len - 1)` for `Found` and `i.saturating_sub(1)` for `NotFound`?

File: vortex-array/src/compute/search_sorted.rs [hunk 2]
```diff
@@ -23,26 +23,59 @@ impl Display for SearchSortedSide {
...
+                }
+            }
+        }
+    }
+
+    /// Convert search result into index suitable for searching non 0 offset ends arrays like ends in RunEndArray
+    pub fn to_ends_index(self, len: usize) -> usize {
+        let idx = self.to_index();
+        if idx == len {
```
Comment (lwwmanning) [inline vortex-array/src/compute/search_sorted.rs:75]: again, maybe better expressed as `min(idx, len - 1)`

File: vortex-array/src/compute/search_sorted.rs [hunk 3]
```diff
@@ -23,26 +23,59 @@ impl Display for SearchSortedSide {
...
 
+    /// Extract index out of search result regardless of whether the value have been found or not
     pub fn to_index(self) -> usize {
         match self {
             Self::Found(i) => i,
             Self::NotFound(i) => i,
         }
     }
+
+    /// Convert search result into an index suitable for index 0 offset ends arrays, like ChunkArray ends
```
Comment (lwwmanning) [inline vortex-array/src/compute/search_sorted.rs]: it's not immediately obvious what you mean by "index 0 offset ends arrays" without looking at the code you reference, maybe expand on the semantics here a bit?

File: vortex-array/src/compute/search_sorted.rs (hunk 1 above)
Comment (robert3005) [reply vortex-array/src/compute/search_sorted.rs:59]: I found often that in practice `if i == 0` is better if min/max doesn't get inlined

**Review by lwwmanning — APPROVED** [review]

**Pull request merged**